assessment/asr_validation/wald2017B/analysis/plots/ASR_validation_plotting.py

```diff
@@ -0,0 +1,238 @@
+import pandas as pd
+import matplotlib.pyplot as plt
+
+# Plot Settings
+plt.style.use('../../../../../scripts/plot_style.mplstyle')
+
+# Global Constants
+AXES = ["x", "y", "z"]
+SEQ = range(1, 7)
+TIME_DIV = 86400
+MEAN_DIV = 0.48
+
+# plotting constants
+COLOR_LINES = ["r-", "b-", "g-"]
+SUBSCRIPTS = ["xx", "yy", "zz"]
+AXES_LIST = ['dispx', 'dispy', 'dispz']
+
+
+def define_dataframes(*args):
+    """Return a list of pandas dataframes."""
+    return [pd.read_csv(fp) for fp in args]
+
+
+def sum_cols(df, col1, col2):
+    """Compute absolute value of two columns from a dataframe."""
+    return abs(df[col1]) + abs(df[col2])
+
+
+def make_disp_sum_func(df):
+    """Return a function initialized with a dataframe."""
+
+    def sum_abs_cols(axis, num):
+        """Sum the columns specified by axis and num."""
+        return sum_cols(df, f"disp_{axis}_p{num}_pos", f"disp_{axis}_p{num}_neg")
+
+    return sum_abs_cols
+
+
+def disp_dict(func):
+    """Return dictionary of key:values for all axes and numbers."""
+    return {f"disp{ax}_p{i}": func(ax, i) for ax in AXES for i in SEQ}
+
+
+def compute_displacements(df):
+    disp_func = make_disp_sum_func(df)
+    new_df = df.assign(**disp_dict(disp_func))
+    return new_df
+
+
+def disp_group(idx):
+    """Utility function used by groupby to group disp{x,y,z} values."""
+    return idx.split("_")[0]
+
+
+def get_displacement_mean(df):
+    """Return dataframe with each row being an axis of displacement."""
+    cols_wanted = [f'disp{i}_p{j}' for i in AXES for j in SEQ]
+    dat = df[cols_wanted].transpose()
+    dat_mean = dat.groupby(by=disp_group).apply(lambda x: x.mean())
+    return dat_mean
+
+
+def row_to_list(df, identifier):
+    """Return row of dataframe as list where index == identifier."""
+    return df.filter(like=identifier, axis=0).squeeze()
+
+
+def simplify(df, identifier):
+    """Return a dataframe with two columns: [time, disp{x|y|z}]."""
+    df_mean = get_displacement_mean(df)
+    temp = {'time': df.time,
+            identifier: row_to_list(df_mean, identifier)}
+    return pd.DataFrame(temp)
+
+
+def main():
+    """Script Driver Function."""
+    # Experiment Data
+    experiment, temp, rh, rebar2 = define_dataframes(
+        "./data/asr_experiment.csv",
+        "./data/ASR_validation_temperature1.csv",
+        "./data/ASR_humidity.csv",
+        "./data/asr_experiment_rebarz.csv"
+    )
+
+    # Simulation Data
+    updated_out, refined_out, time01_out, time02_out, damage_out = define_dataframes(
+        "./data/asr_concrete_block_calibration_creep_updated_out.csv",
+        "./data/asr_concrete_block_calibration_creep_updated_refine_out.csv",
+        "./data/asr_concrete_block_calibration_creep_updated_refine_time1_out.csv",
+        "./data/asr_concrete_block_calibration_creep_updated_refine_time2_out.csv",
+        "./data/asr_concrete_block_validation_rebar_creep_damage_updated_refine_out.csv",
+    )
```

Comment:
Can we define these by looping over all the csv files instead of having to add them individually? This probably will be helpful in the future as we create more assessment cases.